Installation error: version `GLIBC_2.29' not found

I don't know if this is a bug in better-sqlite3 or in snap.

With the following minimal demo package:
```
/tmp/MtBBdL$ cat package.json
{
  "name": "demo",
  "version": "0.0.1",
  "main": "demo.js",
  "dependencies": {
    "better-sqlite3": "^7.0.1"
  }
}
/tmp/MtBBdL$ cat demo.js       
const Database = require('better-sqlite3');
```

I get this error:
```
/tmp/jwbayK$ node demo.js 
/tmp/jwbayK/node_modules/bindings/bindings.js:121
        throw e;
        ^

Error: /snap/core/current/lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/jwbayK/node_modules/better-sqlite3/build/Release/better_sqlite3.node)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1250:18)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
    at bindings (/tmp/jwbayK/node_modules/bindings/bindings.js:112:48)
    at Object.<anonymous> (/tmp/jwbayK/node_modules/better-sqlite3/lib/database.js:9:24)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
```

some system info:
```
Ubuntu 19.10, linux 5.3.0-53-generic x86_64
snap info node:  14/stable:        14.3.0                         2020-05-19 (2752) 30MB classic
Package: libc6                           Version: 2.30-0ubuntu2.1
```

For completeness, of course I executed `npm install` (which compiled better-sqlite from scratch it seems) before running the script. Here's its output:
```
/tmp/jwbayK$ npm install
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/2752/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> better-sqlite3@7.0.1 install /tmp/jwbayK/node_modules/better-sqlite3
> prebuild-install || npm run build-release

prebuild-install WARN install No prebuilt binaries found (target=14.3.0 runtime=node arch=x64 libc= platform=linux)
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/2752/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> better-sqlite3@7.0.1 build-release /tmp/jwbayK/node_modules/better-sqlite3
> node-gyp rebuild --release

make: Entering directory '/tmp/jwbayK/node_modules/better-sqlite3/build'
  TOUCH b857c92884e9598d609f6be182a2595df7a8e00f.intermediate
  ACTION deps_sqlite3_gyp_locate_sqlite3_target_extract_sqlite3 b857c92884e9598d609f6be182a2595df7a8e00f.intermediate
  TOUCH Release/obj.target/deps/locate_sqlite3.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite3/sqlite3.o
  AR(target) Release/obj.target/deps/sqlite3.a
  COPY Release/sqlite3.a
  CXX(target) Release/obj.target/better_sqlite3/src/better_sqlite3.o
In file included from ./src/better_sqlite3.lzz:10,
                 from ../src/better_sqlite3.cpp:4:
/home/mfu/.cache/node-gyp/14.3.0/include/node/node.h:749:51: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>)’ to ‘node::addon_context_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)’} [-Wcast-function-type]
  749 |       (node::addon_context_register_func) (regfunc),                  \
      |                                                   ^
/home/mfu/.cache/node-gyp/14.3.0/include/node/node.h:767:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE_X’
  767 |   NODE_MODULE_CONTEXT_AWARE_X(modname, regfunc, NULL, 0)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mfu/.cache/node-gyp/14.3.0/include/node/node.h:798:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE’
  798 |   NODE_MODULE_CONTEXT_AWARE(NODE_GYP_MODULE_NAME,                     \
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
./src/better_sqlite3.lzz:67:1: note: in expansion of macro ‘NODE_MODULE_INIT’
./src/objects/database.lzz: In static member function ‘static void Database::JS_new(const v8::FunctionCallbackInfo<v8::Value>&)’:
./src/objects/database.lzz:172:21: warning: variable ‘status’ set but not used [-Wunused-but-set-variable]
./src/util/data.lzz: In function ‘v8::Local<v8::Value> Data::GetValueJS(v8::Isolate*, sqlite3_stmt*, int, bool)’:
./src/util/data.lzz:75:92: warning: this statement may fall through [-Wimplicit-fallthrough=]
./src/util/data.lzz:75:197: note: here
./src/util/data.lzz: In function ‘v8::Local<v8::Value> Data::GetValueJS(v8::Isolate*, sqlite3_value*, bool)’:
./src/util/data.lzz:79:81: warning: this statement may fall through [-Wimplicit-fallthrough=]
./src/util/data.lzz:79:175: note: here
  SOLINK_MODULE(target) Release/obj.target/better_sqlite3.node
  COPY Release/better_sqlite3.node
  CC(target) Release/obj.target/test_extension/deps/test_extension.o
  SOLINK_MODULE(target) Release/obj.target/test_extension.node
  COPY Release/test_extension.node
rm b857c92884e9598d609f6be182a2595df7a8e00f.intermediate
make: Leaving directory '/tmp/jwbayK/node_modules/better-sqlite3/build'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN demo@0.0.1 No description
npm WARN demo@0.0.1 No repository field.
npm WARN demo@0.0.1 No license field.

added 68 packages from 38 contributors and audited 68 packages in 61.014s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

(I've also tried using the mentioned `--scripts-prepend-node-path` option, still the same error.)

Any news on this. I am having a similar problem:

`internal/modules/cjs/loader.js:1302
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: /snap/core/current/lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /home/joaquim/Code/Enel/dr-customer-offers-ui/node_modules/node-sass/vendor/linux-x64-83/binding.node)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1302:18)
    at Module.load (internal/modules/cjs/loader.js:1100:32)
    at Function.Module._load (internal/modules/cjs/loader.js:962:14)
    at Module.require (internal/modules/cjs/loader.js:1140:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at module.exports (/home/joaquim/Code/Enel/dr-customer-offers-ui/node_modules/node-sass/lib/binding.js:19:10)
    at Object.<anonymous> (/home/joaquim/Code/Enel/dr-customer-offers-ui/node_modules/node-sass/lib/index.js:14:35)
    at Module._compile (internal/modules/cjs/loader.js:1251:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1272:10)
    at Module.load (internal/modules/cjs/loader.js:1100:32)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
`

I am on Suse Enterprise 12 SP3 and downgrading to 7.1.2 does NOT solve the issue.

npm ERR! g++: error: unrecognized command line option ‘-std=c++14’

Any advice?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -38,7 +38,7 @@ jobs:
   publish:
     if: ${{ github.event_name == 'release' }}
     name: Publishing to NPM
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
     needs: test
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       matrix:
         os:
-          - ubuntu-latest
+          - ubuntu-18.04
           - macos-latest
           - windows-latest
         node:
@@ -54,7 +54,7 @@ jobs:
     strategy:
       matrix:
         os:
-          - ubuntu-latest
+          - ubuntu-18.04
           - macos-latest
           - windows-latest
     name: Prebuild on ${{ matrix.os }}

